### DISCUSSION TITLE
Calling url() on TextColumn breaks table layout

### DISCUSSION THREAD

**Discussion by VicChaos32** [filamentphp/filament#9361] [Help]
I've tried figuring out how to add tel: and mailto: hrefs to customer emails and phone number, but when the table is not basic one column per record, layout breaks.

  ```
  public static function table(Table $table): Table
    {
        return $table
            ->columns([
                Grid::make(4)
                    ->schema([
                        Stack::make([
                            TextColumn::make('name')
                                ->size('2xl')
                                ->color('info')
                                ->icon(fn(Customer $record) => $record->tax_id ? 'heroicon-o-building-office-2' : 'heroicon-o-user'),
                            TextColumn::make('tax_id')
                                ->color('gray')
                                ->prefix(__('TaxID:'))
                        ])->columnSpan(3),

                        Stack::make([
                            TextColumn::make('email'),
                            TextColumn::make('phone')
//                                ->url(fn(Customer $record) => $record->phone ? 'tel:' . $record->phone : null)
                        ])
                            ->alignment(Alignment::End)
                            ->columnSpan(1),
                    ])
            ])
            ->filters([
                //
            ])
            ->actions([
                Tables\Actions\EditAction::make(),
                Tables\Actions\DeleteAction::make()
            ])
            ->bulkActions([
                Tables\Actions\BulkActionGroup::make([
                    Tables\Actions\DeleteBulkAction::make(),
                ]),
            ])->modifyQueryUsing(fn(Builder $query) => $query->latest());
    }
```
With url() comented out:
<img width="1100" alt="Screenshot 2023-10-29 at 17 41 02" src="https://github.com/filamentphp/filament/assets/149320259/89d32ca6-5ed7-43fa-8e7c-195fc187c7b5">
With url() the layout doesn't work anymore:
<img width="1095" alt="Screenshot 2023-10-29 at 17 41 18" src="https://github.com/filamentphp/filament/assets/149320259/c92286da-fb03-4374-bfd0-692e5ed42190">

It only appears to be working as expected with 2 column grid where both Stacks take 1 column. Then there is no difference if url() is commented out or not.
<img width="1108" alt="Screenshot 2023-10-29 at 17 43 56" src="https://github.com/filamentphp/filament/assets/149320259/34e85b27-f7e8-4035-8073-3a81ef07f661">

Am I going the wrong way about it? Is there an alternative layout I'm not seeing?
It would be nice to add href with tel: and mailto: for each customer, and stacking email and phone to save space, something like this, but with actual url() working:
<img width="1107" alt="Screenshot 2023-10-29 at 17 47 07" src="https://github.com/filamentphp/filament/assets/149320259/6dcb25cc-c424-4901-9d5f-680a95d2a1da">

**Comment by VicChaos32** [comment]
I found a workaround that will do, but still takes extra space I'd like to save. Adding a custom action I was able to tack on Call button...

```
           ->actions([
                Tables\Actions\Action::make('Call')
                    ->url(fn(WorkOrder $record) => $record->phone ? 'tel:' . $record->phone : null)
                    ->color('info')
                    ->icon('heroicon-o-phone-arrow-up-right'),
                Tables\Actions\EditAction::make(),
                Tables\Actions\DeleteAction::make(),
            ])
```

<img width="1088" alt="Screenshot 2023-10-29 at 21 05 18" src="https://github.com/filamentphp/filament/assets/149320259/9de14b7d-6ecd-4acb-81f4-fae962034c91">

**Comment by CyberLine** [comment]
Same as https://github.com/filamentphp/filament/issues/9878. I commented the Problem there
